plugins/module_utils/utils.py

```diff
@@ -129,7 +130,10 @@ def is_superset(superset, candidate):
 
 
 def filter_results(results, filter_data) -> list[Any]:
-    return [element for element in results if is_superset(element, filter_data)]
+    results_tmp = copy(results)
+    if type(results) is not list:
+        results_tmp = [results_tmp]
```

Comment:
Do we really want to wrap plain dict into list? It seems more logical for api module to just return plain response from HyperCore API.